worldedit-core/src/main/java/com/sk89q/worldedit/function/factory/Snow.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * WorldEdit, a Minecraft world manipulation toolkit
+ * Copyright (C) sk89q <http://www.sk89q.com>
+ * Copyright (C) WorldEdit team and contributors
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.sk89q.worldedit.function.factory;
+
+import com.sk89q.worldedit.function.Contextual;
+import com.sk89q.worldedit.function.EditContext;
+import com.sk89q.worldedit.function.LayerFunction;
+import com.sk89q.worldedit.function.block.SnowSimulator;
+import com.sk89q.worldedit.session.request.Request;
+
+public class Snow implements Contextual<LayerFunction> {
+
+    private final boolean stack;
+
+    public Snow(boolean stack) {
+        this.stack = stack;
+    }
+
+    @Override
+    public LayerFunction createFromContext(EditContext context) {
+        return new SnowSimulator(Request.request().getEditSession(), this.stack);
```

Comment:
iirc this can use `context.getDestination()`, which on one hand I'm not sure will 100% be an EditSession but on the other I'm not sure SnowSimulator requires any more than a raw Extent.

Reply:
Yeah it just needs a raw extent, fixed.